to.etc.domui/src/main/java/to/etc/domui/dom/html/IPageWithDomuiNavigationCheck.java

```diff
@@ -0,0 +1,16 @@
+package to.etc.domui.dom.html;
+
+import org.eclipse.jdt.annotation.NonNull;
+import to.etc.domui.state.UIGotoContext;
+
+/**
+ * Marks the UrlPage for custom handling of leaving the page with unsaved data, caused by domui navigation.
+ */
+public interface IPageWithDomuiNavigationCheck extends IPageWithNavigationCheck {
+
+	/**
+	 * Implements page specific code how user can handle navigation on modified screen data.
+	 * @return
```

Comment:
Try not to add unused javadoc things. You can switch off generation of those in IntelliJ.